runtime/tests/interpreter/for_test.go

```diff
@@ -255,3 +259,108 @@ func TestInterpretForString(t *testing.T) {
 		value,
 	)
 }
+
+type inclusiveRangeForInLoopTest struct {
+	s, e, step        int8
+	expectedLoopCount int
+	onlySignedTest    bool
+}
+
+func TestInclusiveRangeForInLoop(t *testing.T) {
+	t.Parallel()
+
+	baseValueActivation := sema.NewVariableActivation(sema.BaseValueActivation)
+	baseValueActivation.DeclareValue(stdlib.InclusiveRangeConstructorFunction)
+
+	baseActivation := activations.NewActivation(nil, interpreter.BaseActivation)
+	interpreter.Declare(baseActivation, stdlib.InclusiveRangeConstructorFunction)
+
+	testCases := []inclusiveRangeForInLoopTest{
+		{
+			s:                 0,
+			e:                 10,
+			step:              1,
+			expectedLoopCount: 11,
+		},
+		{
+			s:                 0,
+			e:                 10,
+			step:              2,
+			expectedLoopCount: 6,
+		},
+		{
+			s:                 10,
+			e:                 -10,
+			step:              -2,
+			expectedLoopCount: 11,
+			onlySignedTest:    true,
+		},
+	}
+
+	runTestCase := func(t *testing.T, typ sema.Type, testCase inclusiveRangeForInLoopTest) {
+		t.Run(typ.String(), func(t *testing.T) {
+			t.Parallel()
+
+			code := fmt.Sprintf(
+				`
+					fun test(): Int {
+						let s : %[1]s = %[2]d
+						let e : %[1]s = %[3]d
+						let step : %[1]s = %[4]d
+						let r: InclusiveRange<%[1]s> = InclusiveRange(s, e, step: step)
+
+						var count = 0
+						for c in r {
+							count = count + 1
+						}
+						return count
+					}
+				`,
+				typ.String(),
+				testCase.s,
+				testCase.e,
+				testCase.step,
+			)
+
+			inter, err := parseCheckAndInterpretWithOptions(t, code,
+				ParseCheckAndInterpretOptions{
+					CheckerConfig: &sema.Config{
+						BaseValueActivation: baseValueActivation,
+					},
+					Config: &interpreter.Config{
+						BaseActivation: baseActivation,
+					},
+				},
+			)
+
+			require.NoError(t, err)
+			countValue, err := inter.Invoke("test")
+			require.NoError(t, err)
+
+			AssertValuesEqual(
+				t,
+				inter,
+				interpreter.NewIntValueFromInt64(nil, int64(testCase.expectedLoopCount)),
+				countValue,
+			)
+		})
+	}
+
+	for _, typ := range sema.AllIntegerTypes {
+		for _, testCase := range testCases {
+			if testCase.onlySignedTest {
+				continue
+			}
+			runTestCase(t, typ, testCase)
+		}
```

Comment:
Maybe have two slices `signedTestCases` and `unsignedTestCases` and loop through the two slices separately for the two scenarios. Then checking for `onlySignedTest` is not needed.

Reply:
Done, good suggestion. Thanks